settings/settings-module.php

```diff
@@ -273,29 +271,27 @@ public function save_options() {
 
 		if ( isset( $_POST['module'] ) && $this->module === $_POST['module'] ) {
 			// It's up to the child class to decide which options are saved, whether there are errors or not
-			$post = array_diff_key( $_POST, array_flip( array( 'action', 'module', 'pll_ajax_backend', '_pll_nonce' ) ) );
-			$options = $this->update( $post );
-			$this->options = array_merge( $this->options, $options );
-			update_option( 'polylang', $this->options );
+			$post   = array_diff_key( $_POST, array_flip( array( 'action', 'module', 'pll_ajax_backend', 'pll_ajax_settings', '_pll_nonce' ) ) );
+			$errors = $this->options->merge( $this->prepare_raw_data( $post ) );
```

Comment:
```suggestion
			$posted_options   = array_diff_key( $_POST, array_flip( array( 'action', 'module', 'pll_ajax_backend', 'pll_ajax_settings', '_pll_nonce' ) ) );
			$errors                    = $this->options->merge( $this->prepare_raw_data( $posted_options ) );
```
Let's take the opportunity to set a more suitable name.